ui/test/nuxeo-layout.test.js

```diff
@@ -36,66 +36,100 @@ const base = url.substring(0, url.lastIndexOf('/'));
-    `);
-
-    if (!layout.element) {
-      await waitChanged(layout, 'element');
-    }
-    expect(layout.$$('nuxeo-error').hidden).to.be.true;
-    expect(layout.element.localName).to.equal('dummy-layout');
-    expect(layout.element.text).to.equal('dummy');
-  });
+      const container = layout.shadowRoot.querySelector('#container');
+      const nuxeoError = layout.$$('nuxeo-error');
+      if (!isElementVisible(nuxeoError)) {
+        await waitForAttrMutation(nuxeoError, 'hidden', null);
+      }
+      expect(isElementVisible(container)).to.be.false;
+      expect(isElementVisible(nuxeoError)).to.be.true;
+    });
 
-  test('layout valid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
-  });
+    test('Should display error messages', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href="notfound.html"></nuxeo-layout>
+        `,
+      );
+      await flush();
 
-  test('layout valid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(true);
+      const nuxeoError = layout.$$('nuxeo-error');
+      const errorMessage = nuxeoError.shadowRoot.querySelector('div.message');
+      if (!errorMessage) {
+        await waitForAttrMutation(errorMessage, 'hidden', false);
+      }
+      expect(nuxeoError.code).to.be.equal('404');
+      expect(nuxeoError.message).to.be.equal('Failed to find layout');
+      expect(nuxeoError.url).to.be.equal(layout.href);
+    });
+
+    test('Should hide layout when href is empty', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href=""></nuxeo-layout>
+        `,
+      );
+      expect(isElementVisible(layout)).to.be.false;
+    });
   });
 
-  test('layout invalid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Stamp', () => {
+    test('Should stamp the layout with model', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
+      `);
+      await flush();
+
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      expect(isElementVisible(layout.$$('nuxeo-error'))).to.be.false;
+      expect(layout.element.localName).to.equal('dummy-layout');
+      expect(layout.element.text).to.equal('dummy');
```

Comment:
More than the property itself, we want to see it stamps it correctly, no?
So, I'd say that we want to check for a span with text `dummy`